metrics-operator/controllers/common/providers/dynatrace/dynatrace.go

```diff
@@ -95,6 +95,59 @@ func (d *KeptnDynatraceProvider) EvaluateQuery(ctx context.Context, metric metri
 	return r, b, nil
 }
 
+func (d *KeptnDynatraceProvider) EvaluateQueryForStep(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) ([]string, []byte, error) {
+	if metric.Spec.Range == nil {
+		return nil, nil, fmt.Errorf("spec.range is not defined!")
+	}
+	baseURL := d.normalizeAPIURL(provider.Spec.TargetServer)
+	query := url.QueryEscape(metric.Spec.Query)
+	qURL := baseURL + "v2/metrics/query?metricSelector=" + query + "&from=now-" + metric.Spec.Range.Interval + "/" + metric.Spec.Range.Step
+
+	d.Log.Info("Running query: " + qURL)
```

Comment:
As most of the logic of this function seems to be the same as `EvaluateQuery`, with the difference being the generation of the query and returning `getResultSlice` instead of `getSingleValue` at the end, can you please create a function that performs the request (i.e. creating the `req` object, retrieving the auth token, doing the request and unmarshalling the response payload) that can be used by both `EvaluateQuery` and `EvaluateQueryForStep`?

Reply:
Fixed